lemminx-liberty/src/it/all-locales-schema-and-features-gen-ol-it/src/test/java/LibertyWorkspaceIT.java

```diff
@@ -0,0 +1,183 @@
+        libertyWorkspace.setLibertyInstallationDir(testFolder + "/target/liberty/wlp/bin/tools/ws-schemagen.jar");
+        libertyWorkspace.setLibertyRuntime("ol");
+        libertyWorkspace.setLibertyVersion("ol-25.0.0.2");
+        libertyWorkspace.setLibertyInstalled(true);
+
+        Path schemaGenJarPath = LibertyUtils.findLibertyFileForWorkspace(libertyWorkspace, Paths.get("bin", "tools", "ws-schemagen.jar"));
+        if (schemaGenJarPath != null) {
+            //Generate schema file
+            generateAllLocaleSchemas(libertyWorkspace, schemaGenJarPath);
+        }
+
+        libertyWorkspace.setLibertyInstallationDir(System.getProperty("user.dir") + "/target/liberty/wlp/bin/tools/ws-featurelist.jar");
+        Path featureListJAR = LibertyUtils.findLibertyFileForWorkspace(libertyWorkspace, Paths.get("bin", "tools", "ws-featurelist.jar"));
+        if (featureListJAR != null && featureListJAR.toFile().exists()) {
+            //Generate featurelist file
+            generateAllLocaleFeatureLists(libertyWorkspace, featureListJAR);
+        }
+    }
+
+    public void generateAllLocaleSchemas(LibertyWorkspace libertyWorkspace, Path schemaGenJarPath) {
+        Locale[] localesList = new Locale[]{Locale.ENGLISH, Locale.FRENCH};
+        for (Locale locale : localesList) {
+            generateServerSchemaXsd(libertyWorkspace, schemaGenJarPath, locale);
+        }
+    }
+
+    private void generateServerSchemaXsd(LibertyWorkspace libertyWorkspace, Path schemaGenJarPath, Locale locale) {
+        String projectBaseDir = System.getProperty("user.dir");
+        projectBaseDir = projectBaseDir.replace("/target/it/all-locales-schema-and-features-gen-ol-it", "");
+        Path resourcesSchemaDir = Paths.get(projectBaseDir, "src", "main", "resources", "schema", "xsd", "liberty");
+
+        File schemasDirFile = resourcesSchemaDir.toFile();
+
+        if (!schemasDirFile.exists()) {
+            if (!schemasDirFile.mkdirs()) {
+                LOGGER.warning("Could not create the schemas directory.");
+                return;
+            }
+        }
+
+        String localeTag = locale.toLanguageTag();
+        String baseFilename = "schema";
+        if (libertyWorkspace.isLibertyRuntimeAndVersionSet()) {
+            baseFilename += "-" + libertyWorkspace.getLibertyVersion();
+        }
+
+        File xsdDestFile = new File(schemasDirFile, baseFilename + "-" + localeTag + ".xsd");
+
+        if (!xsdDestFile.exists()) {
+            try {
+                LOGGER.info("Generating schema file from: " + schemaGenJarPath.toString() + " for locale: " + localeTag);
+                String xsdDestPath = xsdDestFile.getCanonicalPath();
+
+                LOGGER.info("Generating schema file at: " + xsdDestPath);
+
+                ProcessBuilder pb = new ProcessBuilder("java", "-jar", schemaGenJarPath.toAbsolutePath().toString(), "--schemaVersion=1.1",
```

Comment:
My main comment for both methods in this test that are generating files is whether we can just use the logic directly from our language server code instead of copying it over here? If you need to change/refactor that code to make it easier to call from here, that is fine.